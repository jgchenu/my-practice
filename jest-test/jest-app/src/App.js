import React from "react";

function App() {
  return (
    <div className="App" data-test="container" title="jgchen">
      hello world !
    </div>
  );
}

export default App;
