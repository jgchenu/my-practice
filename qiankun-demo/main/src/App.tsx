import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '$src/redux/store';
import Routes from '$src/routes';

function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes />
      </BrowserRouter>
    </Provider>
  );
}

export default App;
