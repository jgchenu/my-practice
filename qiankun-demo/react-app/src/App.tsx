import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '$src/redux/store';
import Routes from '$src/routes';

type AppProps = {
  basename?: string;
};
function App(props: AppProps) {
  console.log('props', props);
  return (
    <Provider store={store}>
      <BrowserRouter basename={props.basename}>
        <Routes />
      </BrowserRouter>
    </Provider>
  );
}

export default App;
