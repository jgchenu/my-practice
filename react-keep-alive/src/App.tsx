import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { ConfigProvider } from 'antd';

import { store } from '$src/redux/store';
import Routes from '$src/routes';

import 'normalize.css';
import 'antd/dist/reset.css';
import './style.less';

function App() {
  return (
    <ConfigProvider theme={{ token: { borderRadius: 16, colorPrimary: '#5263e7' } }}>
      <Provider store={store}>
        <BrowserRouter>
          <Routes />
        </BrowserRouter>
      </Provider>
    </ConfigProvider>
  );
}

export default App;
