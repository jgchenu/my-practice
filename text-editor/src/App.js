import React, {
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from "react";
import "./style.less";
import { readFileAsDataUrl } from "./utils";

function App() {
  const [html, setHtml] = useState("");

  const selectedImgRef = useRef(null);

  const execCommand = useCallback((name, args = null) => {
    document.execCommand(name, false, args);
  });

  const handleBold = useCallback(() => {
    execCommand("bold");
  }, []);

  const handleItalic = useCallback(() => {
    execCommand("italic");
  }, []);

  const handleUnderline = useCallback(() => {
    execCommand("underline");
  }, []);

  const handleFormatBlock = useCallback((block = "<p>") => {
    execCommand("formatblock", block);
  }, []);

  const handleAddLink = useCallback(() => {
    let url = window.prompt("请输入链接地址", "https://baidu.com");
    if (!url) {
      return;
    }
    execCommand("createLink", url);
  }, []);

  const handleEditorChange = useCallback((e) => {
    const element = e.target;
    setHtml(element.innerHTML);
  }, []);

  const handleInsertImage = useCallback(async (e) => {
    const target = e.target;
    if (!target) {
      return;
    }
    const file = target.files[0];
    const base64Img = await readFileAsDataUrl(file);
    execCommand("insertImage", base64Img);
    target.value = ""; // 解决同一张图片上传无效的问题
  }, []);

  const imgTagHandler = useCallback((target) => {
    selectedImgRef.current = target;
    target.classList.add("selected");
  }, []);

  const htmlTagHandlerMap = useMemo(() => {
    return {
      img: imgTagHandler,
    };
  }, [imgTagHandler]);

  const clearImgSelected = useCallback(() => {
    const selectedImg = selectedImgRef.current;
    if (!selectedImg) return;
    selectedImg.classList.remove("selected");
  }, []);

  const clearSelected = useCallback(() => {
    clearImgSelected();
  }, []);

  const handleEditorClick = useCallback(
    (e) => {
      clearSelected();
      const target = e.target;
      const tagName = target.tagName?.toLocaleLowerCase();
      const handler = htmlTagHandlerMap[tagName];
      handler && handler(target);
    },
    [htmlTagHandlerMap]
  );

  return (
    <div>
      <h1>editor</h1>
      <div className="nav">
        <button onClick={handleBold}>bold</button>
        <button onClick={handleItalic}>italic</button>
        <button onClick={handleUnderline}>underline</button>
        <button className="image">
          insert image
          <input
            type="file"
            accept="image/gif, image/jpeg, image/png"
            className="file-image"
            onChange={handleInsertImage}
          ></input>
        </button>
        <button onClick={handleAddLink}>add link</button>
        <button onClick={() => handleFormatBlock("<p>")}>P</button>
        <button onClick={() => handleFormatBlock("<h1>")}>H1</button>
      </div>
      <div
        onClick={handleEditorClick}
        contentEditable
        className="editor"
        onInput={handleEditorChange}
      ></div>
      <h1>html content</h1>
      <div>{html}</div>
    </div>
  );
}

export default App;
