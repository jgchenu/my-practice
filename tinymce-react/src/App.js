import React, { useRef } from "react";
import BundledEditor from "./Editor";

export default function App() {
  const editorRef = useRef(null);
  const log = () => {
    if (editorRef.current) {
      console.log(editorRef.current.getContent());
    }
  };
  return (
    <>
      <BundledEditor
        onInit={(evt, editor) => (editorRef.current = editor)}
        initialValue="<p>This is the initial content of the editor.</p>"
        init={{
          height: 500,
          menubar: false,
          plugins: [
            "advlist",
            "anchor",
            "autolink",
            "help",
            "image",
            "link",
            "lists",
            "searchreplace",
            "table",
            "wordcount",
          ],
          toolbar:
            "undo redo | blocks | " +
            "bold italic forecolor | alignleft aligncenter " +
            "alignright alignjustify | bullist numlist outdent indent | " +
            "removeformat | help",
          content_style:
            "body { font-family:Helvetica,Arial,sans-serif; font-size:14px }",
        }}
      />
      <button onClick={log}>Log editor content</button>
    </>
  );
}
