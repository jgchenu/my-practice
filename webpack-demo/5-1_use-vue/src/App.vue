<template>
  <div>
    <h3>{{ message }}</h3>
  </div>
</template>

<script>
export default {
  data() {
    return { message: "Hello World" };
  },
};
</script>

<style>
body {
  background: #000;
}
h3 {
  margin: 40px 0 0;
  color: #42b983;
}
</style>
