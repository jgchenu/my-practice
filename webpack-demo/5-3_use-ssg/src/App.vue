<template>
  <div :class="['main', cls]">
    <h3>{{ message }}</h3>
    <button @click="handleClick">Toggle</button>
  </div>
</template>

<script>
import { ref, computed } from "@vue/reactivity";

export default {
  setup() {
    const isActivity = ref(false);
    const cls = computed(() => (isActivity.value ? "activate" : "deactivate"));
    const handleClick = () => {
      isActivity.value = !isActivity.value;
    };

    return {
      isActivity,
      message: "Hello World",
      cls,
      handleClick,
    };
  },
};
</script>

<style>
h3 {
  color: #42b983;
}
.main {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  padding: 20px 12px;
  transition: background 0.3s linear;
}
.activate {
  background: #000;
}
.deactivate {
  background: #fff;
}
</style>
