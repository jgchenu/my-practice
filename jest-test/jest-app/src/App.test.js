import React from "react";
import { shallow } from "enzyme";
import App from "./App";

// test("renders learn react link", () => {
//   const result = render(<App />);
//   const linkElement = result.getByText(/hello world/i);
//   expect(linkElement).toBeInTheDocument();
// });

it("render without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
  const container = wrapper.find("[data-test='container']");
  // expect(wrapper.find("[data-test='container']")).toBe(1);
  expect(container).toExist(1);
  expect(container).toHaveProp("title", "jgchen");
});
