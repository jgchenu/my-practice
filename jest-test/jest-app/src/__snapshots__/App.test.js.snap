// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
<div
  className="App"
  data-test="container"
  title="jgchen"
>
  hello world !
</div>
`;
